build.gradle

```diff
@@ -23,6 +23,8 @@ repositories {
 dependencies {
     implementation 'org.springframework.boot:spring-boot-starter-jdbc'
     implementation 'org.springframework.boot:spring-boot-starter-web'
+    implementation 'org.springframework.boot:spring-boot-starter-thymeleaf'
+    implementation 'nz.net.ultraq.thymeleaf:thymeleaf-layout-dialect'
```

Comment:
[Thymeleaf Layout Dialect](https://ultraq.github.io/thymeleaf-layout-dialect/) _is_ a third party library, but it's the de facto "one" for doing layout inheritance.  There is no core library, and it's referenced in [Thymeleaf docs](https://www.thymeleaf.org/doc/articles/layouts.html).

Alternative is to have no global layout and pull together / layout Thymeleaf fragments on each page.